aurora/core.py

```diff
@@ -1204,121 +2534,842 @@ def check_conservation(self, plot=True, axs=None, plot_resolutions=False):
-            ax1[1, 1].plot(self.time_out, out["parallel_loss_to_limiter"],
-                           label="Parallel Loss to Limiter")
-            ax1[1, 2].plot(self.time_out, out["edge_loss"], label="Edge Loss")
+            ax2.plot(self.time_plot, all_particles, label="Particles in the plasma", color = blue)
+            ax2.plot(self.time_plot, reservoirs["particles_in_divertor"], label="Particles in the divertor chamber", color = green)
+            if self.namelist["pump_chamber"]:
+                ax2.plot(self.time_plot, reservoirs["particles_in_pump"], label="Particles in the pump chamber", color = light_green)
+            ax2.plot(self.time_plot, reservoirs["particles_stuck_at_main_wall"]+reservoirs["particles_retained_at_main_wall"], label="Particles stored at the main wall", color = light_grey)
+            ax2.plot(self.time_plot, reservoirs["particles_stuck_at_div_wall"]+reservoirs["particles_retained_at_div_wall"], label="Particles stored at the divertor wall", color = grey)
+            ax2.plot(self.time_plot, reservoirs["particles_pumped"], label="Particles pumped away", color = red, linestyle = 'dashed')
+            ax2.plot(self.time_plot, reservoirs["integ_source"], label="Integrated source", color = red, linestyle = 'dotted')
+            ax2.plot(self.time_plot, reservoirs["total"], label="Total particles in the system", color = 'black')
+
+            if (abs((reservoirs["total"][-1] - reservoirs["integ_source"][-1]) / reservoirs["integ_source"][-1]) > 0.1):
+                print("Warning: significant error in particle conservation!")
 
-            ax1[2, 0].plot(self.time_out, out["particles_at_wall"], label="Particles stuck at wall")
-            ax1[2, 1].plot(self.time_out, out["particles_retained_at_wall"],
-                label="Particles retained at wall")
-            ax1[2, 2].plot(self.time_out, out["recycling_from_wall"], label="Wall rec. rate")
+            Ntot = reservoirs["integ_source"][-1]
+            dN = np.trapz((reservoirs["total"] / Ntot - reservoirs["integ_source"] / Ntot) ** 2, self.time_out)
+            dN /= np.trapz((reservoirs["integ_source"] / Ntot) ** 2, self.time_out)
+            print('Particle conservation error %.1f%%' % (np.sqrt(dN) * 100))
+        
+            ax2.set_xlim(self.time_plot[[0, -1]])
+            ax2.set_ylim(0, None)
+            ax2.legend(loc="center left", bbox_to_anchor=(1, 0.5))
+            plt.tight_layout()
 
-            ax1[3, 0].plot(self.time_out, out["recycling_from_divertor"],
-                           label="Divertor rec. rate")
-            ax1[3, 1].plot(self.time_out, out["total"], label="Core impurity particles")
-            for aa in ax1.flatten()[:11]:
-                aa.legend(loc="best").set_draggable(True)
+        if plot:
+            return reservoirs, (ax1, ax2)
+        else:
+            return reservoirs
+  
+    def reservoirs_average_time_traces(self, interval, plot=True, axs=None, plot_resolutions=False):
```

Comment:
@odstrcilt Good idea. However, since these three methods (`reservoirs_time_traces`, `reservoirs_average_time_traces` and `PWI_time_traces`) can do even something more than simply plotting, i.e. they are used to extract several simulated output fields into dictionaries, maybe it is an even better idea to create another file just for these three? Maybe sometehing like output.py? It might make sense also because, combined, these three methods sum to something like 1000 lines together.
(@fsciortino)